OTAnalytics/domain/event.py

```diff
@@ -563,10 +615,90 @@ def get(
 
     def __create_event_list(self, sections: Sequence[SectionId]) -> Iterable[Event]:
         if sections:
-            event_lists = [self._events[section] for section in sections]
-            return list(itertools.chain.from_iterable(event_lists))
+            return list(
+                (
+                    event
+                    for section in sections
+                    for events_by_id in self._events[section].values()
+                    for event in events_by_id
+                )
+            )
         return self.get_all()
 
+    def remove_events_by_road_user_ids(self, road_user_ids: Iterable[TrackId]) -> None:
+        """
+        Removes events associated with the specified road user IDs.
+
+        This method iterates through the provided iterable of road user IDs to delete
+        both non-section and section events related to each ID. After removal, it
+        notifies the subscribers with the updated information about the removed events.
+
+        Args:
+            road_user_ids: An iterable of strings representing the IDs of road users
+                whose events are to be removed.
+
+        Returns:
+            None
+        """
+        removed = []
+        for road_user_id in road_user_ids:
+            removed.extend(
+                self.__remove_non_section_events_by_road_user_id(road_user_id.id)
+            )
+            removed.extend(
+                self.__remove_section_events_by_road_user_id(road_user_id.id)
+            )
+
+        self._subject.notify((EventRepositoryEvent([], removed)))
+
+    def __remove_section_events_by_road_user_id(self, road_user_id: str) -> list[Event]:
+        """
+        Removes all events associated with a specific road user ID from the internal
+        event tracking structure.
+
+        This method iterates through the internal events dictionary, identifies and
+        removes entries corresponding to the given road user ID. Any removed events
+        associated with the specified ID are collected and returned as a list.
+
+        Args:
+            road_user_id (str): The unique identifier of the road user whose events
+                should be removed.
+
+        Returns:
+            list[Event]: A list of Event objects that were associated with the given
+                road user ID and have been removed.
+        """
+        removed = []
+        for track_dict in self._events.values():
+            if track_dict.get(road_user_id) is not None:
```

Comment:
`track_dict` is a `defaultdict`. Would `if road_user_id not in track_dict` also work?
We might also use the walrus operator as in `__remove_non_section_events_by_road_user_id`